### DISCUSSION TITLE
Third party Data Import

### DISCUSSION THREAD

**Discussion by rich86** [hargata/lubelog#445] [Ideas]
Hi,

I'm planning to switch to LubeLogger from an iOS app called Vehicle Smart, which has the ability to automatically download and import data via various government and 3rd party APIs, such as:

> 1. MOT & tax status and MOT history inc. advisories & failures along with
> 2. Full vehicle history (stolen, written off, financed, etc).

For example, here is the UK government API to access MOT data: https://dvsa.github.io/mot-history-api-documentation/

Is this something that may be added to LubeLogger in the future as a feature?

Or, is this sort of functionality something that would be best suited to API Integration, achievable via the API endpoints?

Finally, if it would be possible/recommended to do this via API Integration, is there a way to share or collaborate these integrations with other users of LubeLogger?

Thanks for all the hard work - I look forward to using your software and to supporting the project.

**Comment by hargata** [comment]
@rich86 thanks for reaching out, we believe third party API integrations are best suited to be implemented via API endpoints as there are no way we can feasibly build every single API in the world into LL.

Here are some examples of third party API integrations built and maintained by the community:

https://github.com/mplawner/bouncie-lubelogger-sync
https://github.com/tigattack/lubelogger-fuelio-importer
